airbyte-cdk/python/unit_tests/sources/file_based/file_types/test_csv_parser.py

```diff
@@ -96,3 +103,40 @@ def test_read_and_cast_types(reader_values, expected_rows):
             list(parser._read_and_cast_types(reader, schema, config_format, logger))
     else:
         assert expected_rows == list(parser._read_and_cast_types(reader, schema, config_format, logger))
+
+
+class MockFileBasedStreamReader(AbstractFileBasedStreamReader):
+    def __init__(self, expected_encoding: Optional[str]):
+        self._expected_encoding = expected_encoding
+
+    def open_file(self, file: RemoteFile, mode: FileReadMode, encoding: Optional[str], logger: logging.Logger) -> io.IOBase:
+        assert encoding == self._expected_encoding
```

Comment:
Why are we defining a class `MockFileBasedStreamReader` while we could just "Mock(spec=AbstractFileBasedStreamReader)`? Having the mock would allow us to assert calls (and therefore validate arguments like we're doing with this assert